src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedObjectCreationTests.cs

```diff
@@ -4210,6 +4210,47 @@ public void M2(C c)
                 );
         }
 
+        [Fact]
+        public void TestInConstructorOverloadWithUseSiteError()
+        {
+            var missing = @"public class Missing { }";
+            var missingComp = CreateCompilation(missing, assemblyName: "missing");
+
+            var lib = @"
+public class C
+{
+    public C(Missing m) => throw null;
+    public C(D d) => throw null;
+}
+public class D { }
+";
+            var libComp = CreateCompilation(lib, references: new[] { missingComp.EmitToImageReference() });
+
+            var source = @"
+class D
+{
+    public void M2(C c)
+    {
+        new C(new());
+        new C(default);
+        new C(null);
```

Comment:
This scenarios are not interesting (were not requested) because they are not using target typed new

Reply:
Ah, duh! My bad

Reply:
I reacted too quickly. This test is testing target-typed new (first scenario) and comparing with `default` and `null` behavior with regards to use-site diagnostics.

I can also test `C c = new(null);` which will be ambiguous. Is that what you had in mind?